src/components/AddressAutocomplete.js

```diff
@@ -119,39 +119,62 @@ function AddressAutocomplete(props) {
         ]);
       }
     } else {
-      axios
-        .get(
-          `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${encodeURIComponent(
-            text,
-          )}&${qs.stringify(query)}`,
-          { signal: newController.signal },
-        )
-        .then(response => {
-          const normalizedResults = fuseResults.map(fuseResult => ({
-            ...fuseResult.item,
-            type: 'fuse',
-          }));
-
-          const normalizedPredictions = response.data.predictions.map(
-            prediction => ({
-              ...prediction,
-              type: 'prediction',
-            }),
-          );
-
-          let results = normalizedResults.concat(normalizedPredictions);
-
-          if (normalizedResults.length > 0 && results.length > 5) {
-            results = results.slice(0, 5);
-          }
 
-          setResults(results);
-        })
-        .catch(error => {
-          console.log('AddressAutocomplete; _autocomplete', error);
-        });
+      let normalizedPredictions
+
+      axios.post(
+        'https://places.googleapis.com/v1/places:autocomplete',
+        {
+          input: text,
+          sessionToken: config.sessiontoken,
+          locationRestriction: { "rectangle": {
+            "low" : {
+              "latitude": config.locationrestriction.southWest.split(',')[0],
+              "longitude": config.locationrestriction.southWest.split(',')[1]
+            },
+            "high" : {
+              "latitude": config.locationrestriction.northEast.split(',')[0],
+              "longitude": config.locationrestriction.northEast.split(',')[1]
+            },
+          }},
+          includedPrimaryTypes: ['street_address'],
+          languageCode: config.languageCode
+        },
+        {
+          headers: {"X-Goog-Api-Key": config.key},
+          // signal: controller.signal
+        },
+      ).then(response => {
+        // when there is no response the API returns an empty response
+        const suggestions = response.data.suggestions ? response.data.suggestions  : []
+        
+        normalizedPredictions = suggestions.map(
+          suggestion => ({
+            ...suggestion,
+            description: suggestion.placePrediction.text.text,
+            place_id: suggestion.placePrediction.placeId,
+            type: 'prediction',
+          }),
+        );
+        
+        const normalizedResults = fuseResults.map(fuseResult => ({
+          ...fuseResult.item,
+          type: 'fuse',
+        }));
+
+        let results = normalizedResults.concat(normalizedPredictions);
+
+        if (normalizedResults.length > 0 && results.length > 5) {
+          results = results.slice(0, 5);
+        }
+
+        setResults(results);
+      }).catch(error => {
+        console.error('AddressAutocomplete; _autocomplete', error)
+      })
+    
     }
-  }, 300);
+  }, 400);
```

Comment:
I suspect there is something wrong with `bounce` logic, it sends requests suspiciously often:

![Screenshot 2025-02-20 at 16 41 28](https://github.com/user-attachments/assets/3bc9b1bd-5d67-4491-9f69-d5a8a6fbf3a5)

I also think we can `bounce` more aggressively to decrease the number of requests. Maybe 1 second? (Do the same on the web?)

Reply:
thanks for the testing
the debounce needed obviously to be wrapped in useCallback :laughing: 
i have done some testing and 1s seems a little bit too much to me, i have set to 750ms
i will check on web